webapp/channels/src/components/activity_log_modal/activity_log_modal.tsx

```diff
@@ -47,110 +49,79 @@ export type Props = {
     };
 }
 
-type State = {
-    show: boolean;
-}
-
-export default class ActivityLogModal extends React.PureComponent<Props, State> {
-    static propTypes = {
-
-    };
-
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            show: true,
-        };
-    }
-
-    submitRevoke = (altId: string, e: React.MouseEvent) => {
+const ActivityLogModal = ({
+    currentUserId,
+    sessions,
+    locale,
+    onHide,
+    actions: {
+        getSessions,
+        revokeSession,
+    },
+}: Props) => {
+    const submitRevoke = useCallback((altId: string, e: React.MouseEvent) => {
         e.preventDefault();
         const modalContent = (e.target as Element)?.closest('.modal-content');
         modalContent?.classList.add('animation--highlight');
         setTimeout(() => {
             modalContent?.classList.remove('animation--highlight');
         }, 1500);
-        this.props.actions.revokeSession(this.props.currentUserId, altId).then(() => {
-            this.props.actions.getSessions(this.props.currentUserId);
+        revokeSession(currentUserId, altId).then(() => {
+            getSessions(currentUserId);
         });
-    };
-
-    onShow = () => {
-        this.props.actions.getSessions(this.props.currentUserId);
-    };
-
-    onHide = () => {
-        this.setState({show: false});
-    };
+    }, [currentUserId, revokeSession, getSessions]);
 
-    componentDidMount() {
-        this.onShow();
-    }
+    useEffect(() => {
+        getSessions(currentUserId);
+    }, [currentUserId, getSessions]);
 
-    render() {
-        const activityList = this.props.sessions.reduce((array: JSX.Element[], currentSession, index) => {
-            if (currentSession.props.type === 'UserAccessToken') {
-                return array;
-            }
-
-            array.push(
-                <ActivityLog
-                    key={currentSession.id}
-                    index={index}
-                    locale={this.props.locale}
-                    currentSession={currentSession}
-                    submitRevoke={this.submitRevoke}
-                />,
-            );
+    const activityList = sessions.reduce((array: JSX.Element[], currentSession, index) => {
```

Comment:
0/5 we could put this in a useMemo so we don't have to iterate the full array every time we render.